adapters/kobler/kobler.go

```diff
@@ -0,0 +1,146 @@
+package kobler
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint    string
+	devEndpoint string
+}
+
+const (
+	DEV_BIDDER_ENDPOINT = "https://bid-service.dev.essrtb.com/bid/prebid_server_rtb_call"
+	SUPPORTED_CURRENCY  = "USD"
+)
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint:    config.Endpoint,
+		devEndpoint: DEV_BIDDER_ENDPOINT,
+	}
+
+	return bidder, nil
+}
+
+func (a adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var requestData []*adapters.RequestData
+	var errors []error
+
+	if !contains(request.Cur, SUPPORTED_CURRENCY) {
+		request.Cur = append(request.Cur, SUPPORTED_CURRENCY)
+	}
+
+	for i := range request.Imp {
+		if err := convertImpCurrency(&request.Imp[i], reqInfo); err != nil {
+			errors = append(errors, err)
+			return nil, errors
+		}
+	}
+
+	requestJSON, err := json.Marshal(request)
```

Comment:
Please use our jsonutil helper as a drop in replacement for json. This is a new policy as of PBS 3.0 for performance.